packages/react-charting/src/components/PieChart/Arc/Arc.tsx

```diff
@@ -1,49 +1,131 @@
 import * as React from 'react';
 import * as shape from 'd3-shape';
-import { IArcProps, IArcStyles } from './Arc.types';
-import { classNamesFunction } from '@fluentui/react/lib/Utilities';
+import { IArcProps, IArcState, IArcStyles } from './Arc.types';
+import { classNamesFunction, getId } from '@fluentui/react/lib/Utilities';
 import { getStyles } from './Arc.styles';
 import { convertToLocaleString } from '../../../utilities/utilities';
+import { SVGTooltipText } from '../../../utilities/SVGTooltipText';
+import { select as d3Select } from 'd3-selection';
 
-export class Arc extends React.Component<IArcProps, {}> {
+export class Arc extends React.Component<IArcProps, IArcState> {
   public static defaultProps: Partial<IArcProps> = {
     arc: shape.arc(),
   };
 
-  public state: {} = {};
-
   public static getDerivedStateFromProps(nextProps: Readonly<IArcProps>): null {
     _updateChart(nextProps);
     return null;
   }
 
+  public constructor(props: IArcProps) {
+    super(props);
+    this.state = {
+      isArcFocused: false,
+      id: getId(),
+    };
+  }
+
   public updateChart = (newProps: IArcProps) => {
     _updateChart(newProps);
   };
 
   public render(): JSX.Element {
-    const { color, arc } = this.props;
+    const { arc } = this.props;
     const getClassNames = classNamesFunction<IArcProps, IArcStyles>();
-    const classNames = getClassNames(getStyles, { color });
-    return <path d={arc(this.props.data)} className={classNames.root} onClick={this.props.data?.data.onClick} />;
+    const classNames = getClassNames(props => getStyles(props, this.props.theme), { ...this.props });
+
+    return (
+      <path
+        d={arc(this.props.data)}
+        className={`${this.state.isArcFocused ? classNames.pieRootFocused : classNames.pieRoot}`}
+        onClick={this.props.data?.data.onClick}
+      />
+    );
   }
+
+  protected _onFocus = () => {
+    this.setState({ isArcFocused: true });
+  };
+
+  protected _onBlur = () => {
+    this.setState({ isArcFocused: false });
+  };
 }
 
 export class LabeledArc extends Arc {
+  public constructor(props: IArcProps) {
+    super(props);
+    this.state = {
+      id: getId(),
+    };
+  }
+
   public render(): JSX.Element {
-    const { data, arc, culture } = this.props;
-    const [labelX, labelY] = arc.centroid(data);
-    const labelTranslate = `translate(${labelX}, ${labelY})`;
+    const { data, culture } = this.props;
+    const gap = 4;
+    // placing the labels on the outside arc
+    const [labelX, labelY] = shape.arc().centroid({
+      endAngle: data?.endAngle || 0,
+      startAngle: data?.startAngle || 0,
+      padAngle: data?.padAngle,
```

Comment:
Does padAngle have a default value

Reply:
Arc is the child component of <Pie>. In Pie we're providing the default value to padAngle

![image](https://github.com/microsoft/fluentui/assets/70366079/f34d564f-092d-4efa-8fb0-59aa27fed385)